### ISSUE TITLE
ceph objectstore not created on  openshift: okd 4.4 with rook 1.3

### ISSUE THREAD

**Issue by ibotty** [rook/rook#5264]
**Is this a bug report or feature request?**
* Bug Report

When creating any CephObjectStore (e.g cluster/examples/kubernetes/ceph/object-test.yaml) the operator does not create any rgw pods.  There is no log in the operator pod regarding rgw or a created CephObjectStore.

When a storageclass and a bucketrequest is created there is only (for s3 store):
```
bucket: cephObjectStore service not found: services "rook-ceph-rgw-s3
```

**How to reproduce it (minimal and precise):**
oc apply -f cluster/examples/kubernetes/ceph/object-test.yaml

**Environment**:

* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
bare-metal

* Rook version (use `rook version` inside of a Rook Pod):
rook/ceph:v1.3.1 container image: rook: v1.3.1,go: go1.13.8

* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
Openshift
```
> oc version 
Client Version: 4.5.0-202003312031-25645af
Server Version: 4.4.0-0.okd-2020-01-28-022517
Kubernetes Version: v1.17.1
```

* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):
```
$ ceph -s
  cluster:
    id:     08f51f08-9551-488f-9419-787a7717555e
    health: HEALTH_OK
 
  services:
    mon: 3 daemons, quorum a,d,e (age 18h)
    mgr: a(active, since 3m)
    osd: 3 osds: 3 up (since 18h), 3 in (since 18h)
 
  data:
    pools:   1 pools, 1 pgs
    objects: 3 objects, 0 B
    usage:   3.0 GiB used, 24 TiB / 24 TiB avail
    pgs:     1 active+clean
```

**Comment by leseb** [comment]
What's the CR status? `oc get cephobjectstore -o yaml`?
If you log into the tool box, do you more pools in Ceph too?

Also, the operator definitely has logged this creation, so please look deeper.

**Comment by ibotty** [comment]
```
> oc get cephobjectstore -o yaml | fpaste
Uploading (1.8KiB)...
https://paste.centos.org/view/f8fcf972
> oc logs -n rook-ceph rook-ceph-operator-858df6c77-8n4jj | fpaste 
Uploading (81.3KiB)...
https://paste.centos.org/view/ea8b01e7
```
I (re)created the cephobjectstore a few times in between.  Ceph only has the device_health_metrics pool.

**Comment by leseb** [comment]
To be honest, I have no clue. Can you turn debug logging on?

**Comment by ibotty** [comment]
$ oc logs rook-ceph-operator-7dcb5876cd-fg4m7 |fpaste 
Uploading (94.5KiB)...
https://paste.centos.org/view/fe23f57c

I also noticed that it did not create the CephBlockPool I defined (and I did not find it mentioned in the logs neither).  Maybe there is something fundamentally wrong.

**Comment by travisn** [comment]
A few ideas to investigate:
- Restart the operator to see if it will start reconciling object stores and pools after the fresh start
- Create an object-test.yaml or pool with a different `name` property 
- Do other CRs get created? For example, try creating filesystem-test.yaml or object-user.yaml to see if the operator responds to those events.

**Comment by ibotty** [comment]
I did try any of the things you proposed, but it did not pick any of them up.

I failed to mention that the operator was installed with 1.2.7's operator-openshift.yaml and before doing anything else updated with the upgrade-*.yaml files and the image set to 1.3.0, and then 1.3.1 when that became available.

I have now recreated the operator by using the new operator-openshift.yaml file and it works alright now.

I can provide the files that lead to that situation if you'd like.  My problem is fixed though.
